### PR TITLE
Refactor state components

### LINKED ISSUE #307: Refactor state
- [x] `Editor` should be a component, rather than a page
- [x] State should be a container that is separate from any styles/templates
  - https://github.com/jamiebuilds/unstated

### PR DESCRIPTION
**Pull request by mfix22**
This is just the initial PR. In the future though we can: 
- [x] Separate editor into `Component` (contains container instance) and function (responsible for all DOM elements)
- [x] Remove prop-drilling and have deeper components subscript to editor state
  - [x] Make `<Toolbar/>` (Buttons and settings) `<TextInput>` (Carbon + DnD target) top level with no function props
- [ ] separate state into multiple containers
- [x] Editor -> Carbon, Carbon -> CodeWindow, CodeWindow-> Editor

- Closes #307 
Will reopen another issue with other TODOs

### CODE REVIEW

**Comment by mfix22** [comment]
- Adds linting
  - Closes #275 
- Removes old dependencies

File: components/Coder.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import React from 'react'
```
Comment (jakedex) [inline components/Coder.js]: `Coder` seems kinda confusing to me, how about `CarbonContainer` or something?

File: components/Coder.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import ReadFileDropContainer, { DATA_URL, TEXT } from 'dropperx'
+
+import EditorContainer from '../containers/Editor'
+import Overlay from './Overlay'
+import Carbon from './Carbon'
+
+import { DEFAULT_CODE } from '../lib/constants'
+import { isImage } from '../lib/util'
+
+const to = [EditorContainer]
```
Comment (jakedex) [inline components/Coder.js]: Does this need to be a variable? Might be more readable w/o it...

Comment (mfix22) [reply components/Coder.js]: I just want to make sure that the props is equivalent as to not cause rerenders do to new array constructions

File: components/Coder.js (hunk 1 above)
Comment (mfix22) [reply components/Coder.js]: I want to rename our Carbon to Coder, the current Coder component to Editor, and then the Editor component to Carbon. What do you think?

File: components/Carbon.js [hunk 3]
```diff
@@ -1,279 +1,41 @@
...
-      )
-    }
-
     return (
-      <div id="section">
-        <div id="export-container" ref={ele => (this.exportContainerNode = ele)}>
-          {content}
-          <div id="twitter-png-fix" />
+      <Provider inject={this.inject}>
+        <div id="editor">
```
Comment (jakedex) [inline components/Carbon.js]: Let's use #carbon here

File: components/Editor.js [hunk 4]
```diff
@@ -1,276 +1,48 @@
...
+        )}
+      </ReadFileDropContainer>
     )
   }
-}
 
-function isImage(file) {
-  return file.type.split('/')[0] === 'image'
+  render() {
+    return <Subscribe to={to}>{this.renderPane}</Subscribe>
```
Comment (briandennis) [inline components/Editor.js]: let's rename `to` to something more meaningful

File: components/CodeWindow.js [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+  codeUpdated(newCode) {
+    this.handleLanguageChange(newCode)
+    this.props.updateCode(newCode)
+  }
+
+  handleTitleBarChange(newTitle) {
+    this.props.updateTitleBar(newTitle)
+  }
+
+  handleLanguageChange = debounce(
```
Comment (briandennis) [inline components/CodeWindow.js]: Can this be extracted to the state container?

File: containers/Editor.js [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+class EditorContainer extends Container {
+  constructor(props = {}) {
+    super(props)
+    this.state = {
+      ...DEFAULT_SETTINGS,
+      uploading: false,
+      code: props.content
+    }
+
+    this.save = this.save.bind(this)
```
Comment (briandennis) [inline containers/Editor.js]: are any of these private? If so, maybe specify which are

File: components/CodeWindow.js (hunk 5 above)
Comment (mfix22) [reply components/CodeWindow.js]: Yes. But can we extract it later? there are a bunch of spots we can add new subscriptions and containers, but the goal of this PR is mostly to refactor the top level design and let Carbon be exportable on it's own.

File: containers/Editor.js (hunk 6 above)
Comment (mfix22) [reply containers/Editor.js]: None of them are as of now. Just rename one last one

**Pull request merged**